### PR TITLE
Add User functions

### PR DESCRIPTION
**Pull request by pauljohanneskraft**
# Add User functions

## :recycle: Current situation & Problem
*Link any open issues or pull requests (PRs) related to this PR. Please ensure that all non-trivial PRs are first tracked and discussed in an existing GitHub issue or discussion.*

## :gear: Release Notes 
*Add a bullet point list summary of the feature and possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented or links to the documentation if it appends or changes the public interface.*

## :books: Documentation
*Please ensure that you properly document any additions in conformance to [Spezi Documentation Guide](https://github.com/StanfordSpezi/.github/blob/main/DOCUMENTATIONGUIDE.md).*
*You can use this section to describe your solution, but we encourage contributors to document your reasoning and changes using in-line documentation.* 

## :white_check_mark: Testing
*Please ensure that the PR meets the testing requirements set by CodeCov and that new functionality is appropriately tested.*
*This section describes important information about the tests and why some elements might not be testable.*

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: functions/src/functions/types.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export interface Result<T> {
```
Comment (arkadiuszbachorski) [inline functions/src/functions/types.ts:1]: I think we could make it smarter with union. This way, there is no need for double-checking of `data`/`error` existance. 

Take a look at [this TS REPL](https://www.typescriptlang.org/play/?ssl=12&ssc=1&pln=4&pc=1#code/KYDwDg9gTgLgBDAnmYcBKwDOBXANjAHgBUA+OAXjgG84ATAQxnoC44iAaOYKKaAflbYAdrWAAzAJZDgtOAF84AH2p1GLOMNGTptTt15RWVAMYRRrTDChSA5pwC2WTPRvALV2-LkBYAFB-TIUs4TAhHAAl6EVxuCjgACigsSCC3dCw8QktrIRsSAEoKMio-OAkxBKTMFMxgADp9aEKS3zg2uAB6DrgqmvrGqDgAC25UCUw4CAAjACtgY3gAdwkYIZVTcxCPXIcnFzTszx9WhWBcWupS9q6e5IhUuoYmBGRUEaSyifpcJPpaRDgQnoPAgixkCAgWxyNiubV691qjzUdWMQ2BAEEYPEAKz5UrHY5+IA)

**Review by arkadiuszbachorski — APPROVED** [review]
Looks great, thank you! 🚀

File: functions/src/functions/types.ts (hunk 1 above)
Comment (pauljohanneskraft) [reply functions/src/functions/types.ts:1]: sounds good to me - thank you for the suggestion!

File: .github/workflows/lintbuildandtest.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
       - name: Use Node.js 22.x
         uses: actions/setup-node@v4
         with:
-          node-version: 22.x
```
Comment (pauljohanneskraft) [inline .github/workflows/lintbuildandtest.yml:27]: This line was changed due to: https://github.com/npm/cli/issues/7657

**Pull request merged**